java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/WorkloadIdentityFederationGrantTokenRequestBuilder.java

```diff
@@ -0,0 +1,88 @@
+package co.worklytics.psoxy.gateway.impl.oauth;
+
+import co.worklytics.psoxy.gateway.ConfigService;
+import co.worklytics.psoxy.gateway.RequiresConfiguration;
+import com.google.api.client.http.HttpContent;
+import com.google.api.client.http.UrlEncodedContent;
+import lombok.Getter;
+import lombok.SneakyThrows;
+
+import javax.inject.Inject;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+/**
+ * implementation of a "workload identity" credentials.
+ * This is same as client credentials with the difference that the JWT assertion needs to be externally provided and
+ * claims "aud" and "iss" should be the same between two parts.
+ *
+ * @see <a href="https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow#third-case-access-token-request-with-a-federated-credential">...</a>
+ * as an example of MSFT implementation
+ *
+ * - <a href="https://datatracker.ietf.org/doc/html/rfc7521">...</a>
+ * - <a href="https://datatracker.ietf.org/doc/html/rfc6749#section-4.4.2">...</a>
+ */
+public abstract class WorkloadIdentityFederationGrantTokenRequestBuilder
+        implements OAuthRefreshTokenSourceAuthStrategy.TokenRequestBuilder, RequiresConfiguration {
```

Comment:
Now I see; worth to do some stuff and inherit from `ClientCredentialsGrantTokenRequestBuilder` to reuse properties and claims? Both are the same, except `ClientCredentialsGrantTokenRequestBuilder` builds the JWT with the private key provided meanwhile here it is responsible from implementation do obtain it somewhere